src/Bridge/Doctrine/MongoDbOdm/Paginator.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\MongoDbOdm;
+
+use ApiPlatform\Core\DataProvider\PaginatorInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Doctrine\ODM\MongoDB\Iterator\Iterator;
+use Doctrine\ODM\MongoDB\UnitOfWork;
+
+/**
+ * Decorates the Doctrine MongoDB ODM paginator.
+ *
+ * @experimental
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class Paginator implements \IteratorAggregate, PaginatorInterface
+{
+    /**
+     * @var Iterator
+     */
+    private $mongoDbOdmIterator;
+    /**
+     * @var array
+     */
+    private $pipeline;
+    /**
+     * @var UnitOfWork
+     */
+    private $unitOfWork;
+    /**
+     * @var string
+     */
+    private $resourceClass;
+
+    /** @var \ArrayIterator */
+    private $iterator;
+
+    /**
+     * @var int
+     */
+    private $firstResult;
+    /**
+     * @var int
+     */
+    private $maxResults;
+    /**
+     * @var int
+     */
+    private $totalItems;
+
+    public function __construct(Iterator $mongoDbOdmIterator, UnitOfWork $unitOfWork, string $resourceClass, array $pipeline)
+    {
+        $this->mongoDbOdmIterator = $mongoDbOdmIterator;
+        $this->unitOfWork = $unitOfWork;
+        $this->resourceClass = $resourceClass;
+        $this->pipeline = $pipeline;
+
+        $resultsFacetInfo = $this->getFacetInfo('results');
+        $this->getFacetInfo('count');
+
+        // See https://github.com/alcaeus/mongo-php-adapter#mongocommandcursor
```

Comment:
I'd remove the reference to `mongo-php-adapter` since it's not used at all here. Suggested wording:

> Since the \MongoDB\Driver\Cursor class does not expose information about skip/limit parameters of the query, the values set in the facet stage are used instead.